contributing/checklist.md

```diff
@@ -303,7 +303,8 @@ They should focus on educating thru the catalog’s visual result and the code i
 
 Our users create their views both in code and in Interface Builder. It’s important to support both usages. Almost all components should be able to be added to a view hierarchy thru Interface Builder.
 
-1. UIView subclasses must support initWithCoder along with initWithFrame. The recommended practice is to override both init methods and have them both call a commonInit method with required initialization logic.
+1. UIView subclasses must support initWithCoder along with initWithFrame. The recommended practice is to override both init methods and have them both call a [common*Init](#commonmdcclassinit-if-necessary) method with required initialization logic.
+    1. Remember to call the `common*Init` before unarchiving the values from the decoder.
```

Comment:
For example, you can see [how tricky it is to get the subview restoration right in MDCBottomNavigationItemView](https://github.com/material-components/material-components-ios/blob/develop/components/BottomNavigation/src/private/MDCBottomNavigationItemView.m#L73).  We've even added [unit tests to verify that we don't introduce bugs during decoding](https://github.com/material-components/material-components-ios/blob/develop/components/BottomNavigation/tests/unit/BottomNavigationItemViewCodingTests.m#L75).

Reply:
We follow the pattern set by Apple.  Whether you initWithFrame: or initWithCoder: UIButton will stand up its subviews / label.  What alternative would you propose to this PR?

Reply:
My alternative is to not recommend using `commonInit` globally, but make those kinds of decisions on a case-by-case basis. It may be that a designated initializer does sufficient initialization that a convenience initializer can just do a bit of tweaking afterward.

Reply:
I'm not sure I follow.  Both initWithFrame: and initWithCoder: are designated initializers.

Imagine the following:
In my project I have a storyboard
I drag a UIView into it
I set the class of that view to MDCActivityIndicator

I would expect that when I instantiate that view controller from its nib/storyboard that the view in question would be a complete MDCActivityIndicator.

How does that work in you scenario where initWithCoder: does not instantiate subviews?

Reply:
`-initWithCoder:` may not need to initialize and add subviews. If that work was handled by `UIView`, then the existing subview properties can simply be deserialized from the archive and the view is good-to-go.  However, if there are no such key/value pairs in the archive, then the `-initWithCoder:` will need to create them.  Looking at `MDCBottomNavigationItemView`, I think that particular code can be cleaned-up.  However, imagine this scenario:

I have a UIView subclass with a single, optional subview, `myView`. In `-initWithFrame:` I create and add that subview to myself.  At some point along the way, `myView` is removed from my subviews and the property nilled-out.  When I encode, I can encode the `nil` value, but the archive will contain the key.  When I decode, I can check that the key exists, then decode an explicit `nil` value. If I then run my `commonInit`, I might miss that the property *should* be`nil` and then accidentally recreate it.